### ISSUE TITLE
metaschema validate behaviour has changed since 2.2.0

### ISSUE THREAD

**Issue by wandmagic** [metaschema-framework/metaschema-java#254]
### Describe the bug

if this workflow is run with any version higher than 2.2.0

external constraints in metaschema validate command are ignored

https://github.com/GSA/fedramp-automation/blob/develop/.github/workflows/run-lint.yml

### Who is the bug affecting

metaschema developers

### How do we replicate this issue

1) fork this repo fedramp-automation repo
2) change version to 2.3.1
3) make a constraint not follow the rules (remove help link prop)
4) run action (see false positive and no constraint extension loaded)

### Expected behavior (i.e. solution)

1) fork this repo
2) change version to 2.3.1
3) make a constraint not follow the rules (remove help link prop)
4) run action (see constraint errors)

### Other comments

something is up with the external constraint loaders, my suggestion is we make a unit test that runs the CLI with external constraints on the metaschema validate command

**Comment by david-waltermire** [comment]
@aj-stein-gsa or @wandmagic Do you want to look into a fix for this?

**Comment by david-waltermire** [comment]
@wandmagic I am going to need your help reproducing this.

**Comment by wandmagic** [comment]
this command will say that the file is valid
```
npx oscal use  2.3.1
npx oscal-cli metaschema validate https://raw.githubusercontent.com/gsa/fedramp-automation/refs/heads/develop/src/validations/constraints/oscal-external-constraints.xml -c https://raw.githubusercontent.com/gsa/fedramp-automation/refs/heads/develop/src/validations/styleguides/fedramp-constraint-style.xml
```


this command will say that the file is invalid
```
npx oscal use  2.2.0
npx oscal-cli metaschema validate https://raw.githubusercontent.com/gsa/fedramp-automation/refs/heads/develop/src/validations/constraints/oscal-external-constraints.xml -c https://raw.githubusercontent.com/gsa/fedramp-automation/refs/heads/develop/src/validations/styleguides/fedramp-constraint-style.xml
```

**Comment by wandmagic** [comment]
this is confirmed fixed in 2.4.0 SNAPSHOT.

nice work!
